### PR TITLE
Fixes #13459 - libvirt name coding

### PR DESCRIPTION
**Pull request by ezr-ondrej**
If libvirt has some special characters in volume/network names it breaks rendering.

Libvirt returns ASCII-8BIT encoded string with UTF8 encoded characters in it.
It's probably a bug in ruby-libvirt, but this fixes thigs on our side.

ASCII-8BIT string then forces the renderer to switch the buffer encoding to ASCII-8BIT.
Once there is another string that needs UTF-8 (owner name eg.) it fails to concat that string to buffer.

### CODE REVIEW

**Comment by lzap** [comment]
Czech locale?!? :-)

Unfortunately error page suffers from this as well, when there is a problem with libvirt, it tries to render the error message and then it ends up with nasty 500 error. We can't workaround that I think. Also this is quite nasty. I'd prefer fixing this in https://gitlab.com/libvirt/libvirt-ruby to be honest.

I quickly looked into this and I can reproduce this:

```ruby
  require 'libvirt'
  
 conn = Libvirt::open("qemu:///system")
  begin
    dom = conn.lookup_domain_by_name("_a_non_existing_domain")
    puts dom.xml_desc
  rescue Libvirt::RetrieveError => e
    puts e.message.encoding
  end
```

Prints indeed ASCII-8BIT

**Comment by lzap** [comment]
Let's try but I think given the issue number this might not be the right place: https://gitlab.com/libvirt/libvirt-ruby/-/issues/1

**Comment by lzap** [comment]
PR to fix this in ruby client library: https://gitlab.com/libvirt/libvirt-ruby/-/merge_requests/3/diffs

Let's hold this if we are able to fix this.

**Comment by ezr-ondrej** [comment]
Thanks a lot for the effort! I'm definitelly up for fixing it at it's source rather then on our side :+1:

**Comment by ezr-ondrej** [comment]
The PR to ruby-libvirt has been merged, let's hope for quick release :)

**Comment by lzap** [comment]
There is not gonna be a quick release, I was told. Let's merge this as this won't hurt even when fixed library is used.

**Review by lzap — APPROVED** [review]
Merge this as a temporary workaround until the library is fixed.

**Pull request merged**
